File: SharePoint/CSOM/CSOM slides/materials/exercise1/Demos/CS/ManagedCodeDemo/ManagedCodeDemo/Form1.cs
using Microsoft.SharePoint.Client;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using System.Web.Script.Serialization;
using System.Windows.Forms;
using System.Xml.Linq;

namespace ManagedCodeDemo
{
    public partial class Form1 : System.Windows.Forms.Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void LoadButton_Click(object sender, EventArgs e)
        {
            var siteUrl = "http://localhost/sites/demo";
            using (var context = new ClientContext(siteUrl))
            {
                var web = context.Web;
                context.Load(web);
                context.Load(web.Lists);
                context.ExecuteQuery();
                ResultsListBox.Items.Add(web.Title);
                ResultsListBox.Items.Add(web.Lists.Count);
            }
        }

        private void LoadQueryButton_Click(object sender, EventArgs e)
        {
            var siteUrl = "http://localhost/sites/demo";
            using (var context = new ClientContext(siteUrl))
            {
                var web = context.Web;

                var query = from list in web.Lists
                            where list.Hidden == false &&
                                    list.ItemCount > 0
                            select list;
                var lists = context.LoadQuery(query);
                context.ExecuteQuery();

                ResultsListBox.Items.Add(lists.Count());
            }
        }

        private void RestXmlButton_Click(object sender, EventArgs e)
        {
            var url = "http://localhost/sites/demo/_api/Web/";
            var client = new WebClient();
            client.UseDefaultCredentials = true;
            var xml = client.DownloadString(url);

            var doc = XDocument.Parse(xml);
            XNamespace ds = "http://schemas.microsoft.com/" +
                "ado/2007/08/dataservices";
            var titles = doc.Descendants(ds + "Title");
            var title = titles.First().Value;

            ResultsListBox.Items.Add(title);
        }

        private void RestJsonButton_Click(object sender, EventArgs e)
        {
            var url = "http://localhost/sites/demo/_api/Web/";
            var client = new WebClient();
            client.UseDefaultCredentials = true;
            client.Headers[HttpRequestHeader.Accept] =
                "application/json;odata=verbose";
            var json = client.DownloadString(url);

            var ser = new JavaScriptSerializer();
            dynamic item = ser.Deserialize<object>(json);

            ResultsListBox.Items.Add(item["d"]["Title"]);
        }
    }
}

File: SharePoint/CSOM/sharepoint-2013-client-object-model-rest/materials/1-sharepoint-2013-client-object-model-rest-m1-intro-exercise-files/Demos/VB/ManagedCodeDemo/ManagedCodeDemo/Form1.vb
Imports Microsoft.SharePoint.Client
Imports System.Net
Imports System.Web.Script.Serialization

Public Class Form1
    Private Sub LoadButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles LoadButton.Click
        Dim siteUrl = "http://localhost/sites/demo"
        Using context = New ClientContext(siteUrl)
            Dim web = context.Web
            context.Load(web)
            context.Load(web.Lists)
            context.ExecuteQuery()
            ResultsListBox.Items.Add(web.Title)
            ResultsListBox.Items.Add(web.Lists.Count)
        End Using
    End Sub

    Private Sub LoadQueryButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles LoadQueryButton.Click
        Dim siteUrl = "http://localhost/sites/demo"
        Using context = New ClientContext(siteUrl)
            Dim web = context.Web

            Dim query = From list In web.Lists
                        Where list.Hidden = False AndAlso
                            list.ItemCount > 0
                        Select list
            Dim lists = context.LoadQuery(query)
            context.ExecuteQuery()

            ResultsListBox.Items.Add(lists.Count())
        End Using
    End Sub

    Private Sub RestXmlButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles RestXmlButton.Click
        Dim url = "http://localhost/sites/demo/_api/Web/"
        Dim client = New WebClient()
        client.UseDefaultCredentials = True
        Dim xml = client.DownloadString(url)

        Dim doc = XDocument.Parse(xml)
        Dim ds As XNamespace = "http://schemas.microsoft.com/" & "ado/2007/08/dataservices"
        Dim titles = doc.Descendants(ds + "Title")
        Dim title = titles.First().Value

        ResultsListBox.Items.Add(title)
    End Sub

    Private Sub RestJsonButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles RestJsonButton.Click
        Dim url = "http://localhost/sites/demo/_api/Web/"
        Dim client = New WebClient()
        client.UseDefaultCredentials = True
        client.Headers(HttpRequestHeader.Accept) =
            "application/json;odata=verbose"
        Dim json = client.DownloadString(url)

        Dim ser = New JavaScriptSerializer()
        Dim item = ser.Deserialize(Of Object)(json)

        ResultsListBox.Items.Add(item("d")("Title"))
    End Sub
End Class
